Inline validation in signup and login

AlfiyaSiddique opened the pull request:
## Description
Added inline validation in signup and login forms.

### Solves issue #590 Real-time validation in login and signup.

## Sample Output Video:

https://github.com/narayan954/dummygram/assets/86224794/2be2798b-5338-4b57-8645-b98d53b0ac99

narayan954 reviewed (changes requested):
I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me.
![image](https://github.com/narayan954/dummygram/assets/77617189/b45380c4-447d-4136-967b-89b7e1316056)
Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that

AlfiyaSiddique commented on src/pages/Signup/index.jsx:
```diff
@@ -243,25 +253,50 @@ const SignupScreen = () => {
+              handleError(e.target.name, e.target.value);
+            }}
+            className={
+              error.emailError? "username-not-available": null
```
I used this class to make the borders red when an error exists, so I guess to make it readable and generalize it we can make rename the class as error-border, what do you say??

narayan954 replied:
yes that'd be great

AlfiyaSiddique commented:
> I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me. ![image](https://user-images.githubusercontent.com/77617189/250267680-b45380c4-447d-4136-967b-89b7e1316056.png) Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that

About the login password, If the validation exists it will provide immediate feedback to the user to type the correct password in case they typed incorrect password validation-wise,  Also if you think there's no need for that then let me know I'll do as you say.

And about restricting form submission ,  I'll fix it as soon as possible and make  a commit.

narayan954 commented:
> > I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me. ![image](https://user-images.githubusercontent.com/77617189/250267680-b45380c4-447d-4136-967b-89b7e1316056.png) Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that
> 
> About the login password, If the validation exists it will provide immediate feedback to the user to type the correct password in case they typed incorrect password validation-wise, Also if you think there's no need for that then let me know I'll do as you say.
> 
> And about restricting form submission , I'll fix it as soon as possible and make a commit.

you may remove the password validation at login, because old users won't be able to login.

AlfiyaSiddique commented:
> > > I doubt we need to validate login form for passwords, maybe I'm incorrect with that.You can tell me. ![image](https://user-images.githubusercontent.com/77617189/250267680-b45380c4-447d-4136-967b-89b7e1316056.png) Also, there's one critical problem. You're validating and showing errors but not restricting any user from login/signup. we can still signup/login along with errors. Please fix that
> > 
> > 
> > About the login password, If the validation exists it will provide immediate feedback to the user to type the correct password in case they typed incorrect password validation-wise, Also if you think there's no need for that then let me know I'll do as you say.
> > And about restricting form submission , I'll fix it as soon as possible and make a commit.
> 
> you may remove the password validation at login, because old users won't be able to login.

Okay, will do it soon. 👍🏻

AlfiyaSiddique commented:
Hello @narayan954 

Implemented the changes you asked for, my new commits contain the following changes:
1. Rename the class username-not-allowed to error-border.
2. Implemented the code for preventing the form from getting submitted if errors exist.
3. Remove the password validation of the login form.

narayan954 commented:
> Hello @narayan954
> 
> Implemented the changes you asked for. My new commits contain the following changes:
> 
> 1. Rename the class **username-not-available** to **error-border**.
> 2. Implemented the code for preventing the form from getting submitted if errors exist.
> 3. Remove the password validation of the login form.

Thanks for fixing them, reviewing

narayan954 reviewed (approved):
looks good to me!
